src/main/java/seedu/duke/command/MarkAsDoneCommand.java

```diff
@@ -0,0 +1,49 @@
+package seedu.duke.command;
+
+import seedu.duke.data.AvailableModulesList;
+import seedu.duke.data.ModuleList;
+import seedu.duke.data.SelectedModulesList;
+import seedu.duke.data.SemModulesList;
+import seedu.duke.module.Module;
+import seedu.duke.ui.Ui;
+
+import java.sql.SQLOutput;
+
+public class MarkAsDoneCommand extends Command {
+    public static final String COMMAND_WORD = "done";
+    private String moduleName;
+    private String semester;
+
+    /**
+     * marks the module in a semester in the selectedList as done.
+     * @param moduleName : name of the module that the user wants to mark as done.
+     * @param semester : Semester that the user wants to mark as done.
+     */
+    public MarkAsDoneCommand(String moduleName, String semester) {
+        super();
+        this.moduleName = moduleName;
+        this.semester = semester;
+    }
+
+    @Override
+    public void execute(SelectedModulesList selectedModulesList, ModuleList availableModulesList) {
+        markAsDoneCommand(selectedModulesList);
+        Ui.showDoneMessage();
+    }
+
+    private void markAsDoneCommand(SelectedModulesList selectedModulesList) {
+        for (SemModulesList sem: selectedModulesList) {
+            if (sem.getSem().equals(semester)) {
+                for (Module module: sem) {
+                    System.out.println(module.getName());
+                    System.out.println(module.toString());
+                    if (module.getName() != null) {
+                        if (module.getName().equals(moduleName)) {
+                            module.setAsDone();
+                        }
+                    }
```

Comment:
> Added MarkAsDone class for String, but not added for ID yet. Currently i'm not sure how to allow MarkAsDone class to differentiate between moduleId and moduleName efficiently

I think we just need to change the code into:
```
if (module.getName().equals(moduleName) || module.getId().equals(moduleName)) {
          module.setAsDone();
}
```
(ps: `moduleName` here can also be ID if user input a module ID [maybe we can change the variable name of `moduleName`]. And in Module class, we can initialize `name` and `id` as `""` so we can omit `module.getName() != null` here.）

Reply:
Okay, let me change that tonight! Will update the code again later on!

Reply:
Okay, thank you!

Reply:
Okay, thank you!